cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamHandler.java

```diff
@@ -182,17 +181,16 @@ public void getInfo(HttpRequest request, HttpResponder responder,
   public void create(HttpRequest request, HttpResponder responder,
                      @PathParam("namespace-id") String namespaceId,
                      @PathParam("stream") String stream) throws Exception {
-
-    String accountID = getAuthenticatedAccountId(request);
-    Id.Stream streamId = Id.Stream.from(accountID, stream);
-
     // Verify stream name
-    if (!isValidName(stream)) {
+    if (!Id.isId(stream)) {
```

Comment:
isn't this functionally the same? Now I'm confused why it's returning a 500 and not 400.